File: Singleton/TypeScript/src/Singleton.ts
export class Singleton {
  private static uniqueInstance: Singleton;

  private constructor() {}

  public static getInstance(): Singleton {
    if (!this.uniqueInstance) {
      this.uniqueInstance = new Singleton();
    }

    return this.uniqueInstance;
  }

  getDescription(): string {
    return "Hello, I'm one of a kind :)";
  }
}

File: Singleton/Java/src/Singleton.java
package eu.svetlin.design.patterns.singleton;

public class Singleton {
    private volatile static Singleton uniqueInstance;

    private Singleton(){}

    public String getDescription(){
        return "Hello, I'm one of a kind :)";
    }

    public static Singleton getInstance(){
        if(uniqueInstance == null){
            synchronized (Singleton.class){
                if(uniqueInstance == null){
                    uniqueInstance = new Singleton();
                }
            }
        }
        return uniqueInstance;
    }
}
